packages/knip/src/plugins/angular/index.ts

```diff
@@ -44,24 +45,37 @@ const resolveConfig: ResolveConfig<AngularCLIWorkspaceConfiguration> = async (co
       );
       const productionEntriesByOption: EntriesByOption =
         entriesByOptionByConfig.get(PRODUCTION_CONFIG_NAME) ?? new Map();
-      const normalizePath = (path: string) => join(cwd, path);
+      const isBuildTarget = targetName === BUILD_TARGET_NAME;
+      const maybeExternal = (option: string) => option === 'polyfills';
+      const toInput = (specifier: string, opts: { isProduction: boolean; maybeExternal: boolean }): Input => {
+        const normalizedPath = join(cwd, specifier);
+        // 👇 `isInternal` will report `false` for specifiers not starting with `.`
+        //    However, relative imports are usually specified in `angular.json` without `.` prefix
+        //    Hence checking also that file doesn't exist before considering it external
+        if (opts.maybeExternal && !isInternal(specifier) && !existsSync(normalizedPath)) {
+          return toDeferResolve(specifier);
+        }
+        return opts.isProduction ? toEntry(normalizedPath) : toProductionEntry(normalizedPath);
```

Comment:
Is this ternary intentionally "swapped"?

Reply:
Whoops 😓 Absolutely not, it's a big brain fart 

Fixed in 8e917eef0b2de4cd2cb5ea484669bb00125499f6